File: LeetCode/Easy/2176. Count Equal and Divisible Pairs in an Array/2025-06-02_Accepted.py
class Solution:
  def countPairs(self, nums: List[int], k: int) -> int:
    idx_map = {}
    for i, num in enumerate(nums):
      if num not in idx_map:
        idx_map[num] = []
      idx_map[num].append(i)
    
    ans = 0
    for num in idx_map:
      ans += self.count(idx_map[num], k)
    return ans
  
  def count(self, idxs: List[int], k: int) -> int:
    result = 0
    for i in range(len(idxs)):
      for j in range(i + 1, len(idxs)):
        if (idxs[i] * idxs[j]) % k == 0:
          result += 1
    return result

File: LeetCode/Easy/2180. Count Integers With Even Digit Sum/2025-06-02_Accepted.cpp
class Solution {
public:
  int countEven(int num) {
    int ans = 0;
    for (int i = 1; i <= num; i++) {
      if (isEven(i)) {
        ans++;
      }
    }
    return ans;
  }

private:
  bool isEven(int x) {
    int sum = 0;
    while (x > 0) {
      sum += x % 10;
      x /= 10;
    }
    return sum % 2 == 0;
  }
};